.gitpod.Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM gitpod/workspace-base
+
+ENV NODE_VERSION=14.19.0
+
+# Installation Prep
+RUN sudo apt-get update && sudo apt-get install -y \
+    libusb-1.0-0-dev \
+    libxshmfence1 \
+    libglu1 \
+    libgtk-3-dev \
+    libgconf2-dev \
+    libxss1 \
+    libnss3-dev \
+    libasound2 \
+    libudev-dev \
+    unzip \
+    cmake
+
+USER gitpod
+
+RUN curl -fsSL https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | PROFILE=/dev/null bash \
+    && bash -c ". .nvm/nvm.sh \
+        && nvm install v${NODE_VERSION} \
+        && nvm alias default v${NODE_VERSION} \
+        && npm install -g typescript yarn pnpm node-gyp" \
```

Comment:
I'm not convinced that they are useful.

I find that typescript have too many breaking changes for that to work. I think it was 4.5 that made it so that `} catch (e) {` had to be changed to `} catch (e: any) {` or `unknown` could be used instead of `any`, and there have been other things in previous releases.
Every module which needs typescript has it as a devDependency already, so this global version should never be needed.

If node-gyp is needed to build something, then it needs to be defined as a project dependency somewhere, or it will cause the general companion builds to fail. And it will fail on my machine, and on the machine of almost every other module developer.